packages/reactive-element/src/css-tag.ts

```diff
@@ -182,7 +182,13 @@ export const adoptStyles = (
         style.setAttribute('nonce', nonce);
       }
       style.textContent = (s as CSSResult).cssText;
-      renderRoot.appendChild(style);
+      let renderInto;
+      if (renderRoot instanceof Document) {
+        renderInto = renderRoot.head;
```

Comment:
Is this necessary? Two thoughts:

1. Putting the check for a document in the initial render pass for every element is unfortunate
2. This changes the behavior vs native adoptedStyleSheets which always come after document.styleSheets in the CSS cascade. Appending to a ShadowRoot is a close approximation when we assume that a lit-like template has been set up first and nothing will be added after the styles... Document may well have styles in the body though. So if we kept a check I think I'd append to document.documentElement. Those styles till apply, but they're outside of `<body>` and likely to stay last in most pages.

Reply:
Oh and with documentElement I think we could do: `(renderRoot.documentElement ?? renderRoot).append(style)`

Reply:
cc @kevinpschaaf